src/character.h

```diff
@@ -61,14 +61,49 @@ enum fatigue_levels {
     MASSIVE_FATIGUE = 1000
 };
 
+struct layer_details {
+
+    std::vector<int> pieces;
+    int max = 0;
+    int total = 0;
+
+    void reset();
+    int layer( const int encumbrance );
+
+    bool operator ==( const layer_details &rhs ) const {
+        return max == rhs.max &&
+               total == rhs.total &&
+               pieces == rhs.pieces;
+    }
+};
+
 struct encumbrance_data {
     int encumbrance = 0;
     int armor_encumbrance = 0;
     int layer_penalty = 0;
+
+    std::array<layer_details, static_cast<int>( layer_level::MAX_CLOTHING_LAYER )>
+    layer_penalty_details;
+
+    void layer( const layer_level level, const int emcumbrance ) {
+        layer_penalty += layer_penalty_details[static_cast<size_t>( level )].layer( emcumbrance );
```

Comment:
Why the sudden cast to `size_t`? New code above casts to `int` when the value is used as index or size of an array (note that `std::array` uses `std::size_t` as index/size, not `int`, same for its `operator[]`). At least be consistent.

Reply:
You _are_ correct....  This whole thing is a squashed commit, so subsequent lines of code might have been written either at 3am or 5pm. I plead alcohol and/or brainfarts. 

`size_t` it'll be.